azurerm/helpers/azure/batch_pool.go

```diff
@@ -195,11 +222,56 @@ func ExpandBatchPoolStartTask(list []interface{}) (*batch.StartTask, error) {
 		return nil, fmt.Errorf("Error: either auto_user or user_name should be speicfied for Batch pool start task")
 	}
 
+	resourceFileList := startTaskValue["resource_file"].([]interface{})
+	resourceFiles := make([]batch.ResourceFile, 0)
+	for _, resourceFileValueTemp := range resourceFileList {
+		resourceFileValue := resourceFileValueTemp.(map[string]interface{})
+		resourceFile := batch.ResourceFile{}
+		if autoStorageContainerNameValue, ok := resourceFileValue["auto_storage_container_name"]; ok {
```

Comment:
Minor, and not required to merge but this would be little cleaner:

```go
		if v, ok := resourceFileValue["auto_storage_container_name"].(string); ok && v != ""{
			resourceFile.AutoStorageContainerName = &v
		}
```